static/app/components/integrationExternalMappingForm.tsx

```diff
@@ -4,127 +4,153 @@ import capitalize from 'lodash/capitalize';
 import pick from 'lodash/pick';
 
 import {t, tct} from 'sentry/locale';
-import {ExternalActorMapping, Integration, Organization} from 'sentry/types';
+import {ExternalActorMapping, Integration} from 'sentry/types';
+import {getExternalActorEndpointDetails} from 'sentry/utils/integrationUtil';
 import {FieldFromConfig} from 'sentry/views/settings/components/forms';
 import Form from 'sentry/views/settings/components/forms/form';
+import FormModel from 'sentry/views/settings/components/forms/model';
 import {Field} from 'sentry/views/settings/components/forms/type';
 
-type Props = Pick<Form['props'], 'onSubmitSuccess' | 'onCancel'> &
-  Partial<Pick<Form['props'], 'onSubmit'>> & {
-    organization: Organization;
-    integration: Integration;
-    mapping?: ExternalActorMapping;
-    type: 'user' | 'team';
-    baseEndpoint?: string;
-    sentryNamesMapper: (v: any) => {id: string; name: string}[];
-    url: string;
-    onResults?: (data: any) => void;
-  };
+type Props = Pick<Form['props'], 'onCancel' | 'onSubmitSuccess' | 'onSubmitError'> & {
+  integration: Integration;
+  mapping?: ExternalActorMapping;
+  type: 'user' | 'team';
+  getBaseFormEndpoint: (mapping?: ExternalActorMapping) => string;
+  sentryNamesMapper: (v: any) => {id: string; name: string}[];
+  dataEndpoint: string;
+  onResults?: (data: any, mappingKey?: string) => void;
+  isInline?: boolean;
+  mappingKey?: string;
+};
 
 export default class IntegrationExternalMappingForm extends Component<Props> {
+  model = new FormModel();
```

Comment:
We need to have control of the form model in order to update the `apiMethod` and `apiEndpoint` depending on the contents of the form.

For example, if the form is submitted inline, it needs to be a PUT request, but the creation modal submits to a POST endpoint. And again, depending on the team selected, the endpoint needs to be updated, so we have to change `model.options.apiEndpoint`.